### ISSUE TITLE
[Docs] Exploring code from the docs is cumbersome

### ISSUE THREAD

**Issue by ArturNiederfahrenhorst** [ray-project/ray#48128]
### Description

It’s cumbersome to explore our code from the docs. Basically, you can click to view the source code of a component. But from there, you can’t explore further (which is often needed).

Example:

- Go to https://docs.ray.io/en/latest/tune/api/doc/ray.tune.Tuner.html 
- Click [source] to go to source code
-Click on `AirEntrypoint`  (nothing happens)
- This is where you have to to go GH or another tool and manually find the code there

**Comment by dstrodtman** [comment]
Agreed. This dovetails with currently funded work to simplify code and config extraction for humans and agents.
